okta/komand_okta/actions/get_factors/action.py

```diff
@@ -0,0 +1,49 @@
+import komand
+from .schema import GetFactorsInput, GetFactorsOutput, Input, Output, Component
+# Custom imports below
+import requests
+import json
+from komand.exceptions import PluginException
+
+
+class GetFactors(komand.Action):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name='get_factors',
+            description=Component.DESCRIPTION,
+            input=GetFactorsInput(),
+            output=GetFactorsOutput())
+
+    def run(self, params={}):
+        user_id = params.get(Input.USER_ID)
+        if not user_id:
+            raise PluginException(cause='user_id is required')
+
+        try:
+            okta_url = self.connection.okta_url
+            url = requests.compat.urljoin(okta_url, f'/api/v1/users/{user_id}/factors')
+            response = self.connection.session.get(url)
+            if response.status_code not in range(200, 299):
+                raise PluginException(
+                    cause="Received HTTP %d status code from Okta. Please verify your Okta server status and "
+                          "try again.",
+                    assistance="If the issue persists please contact support.",
+                    data=f"{response.status_code}, {response.text}"
+                )
+            data = response.json()
+
+            return {Output.FACTORS: data}
+
+        except json.decoder.JSONDecodeError:
+            raise PluginException(
+                cause="Received an unexpected response from Okta ",
+                assistance="(non-JSON or no response was received).",
+                data=response.text
+            )
+        except TypeError as e:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON)
+        except KeyError as e:
+            raise PluginException(cause='Key error has occured: %s' % e)
```

Comment:
I would opt for something a little more helpful here if you can. And use the `assistance` argument.

cause="An error occurred while retrieving data from the Okta API.", assistance=e

Same with below

Reply:
Adding on to Jon's comment, when thinking about an error message you need to think about it from the perspective of the user. A "key error" means nothing to them - what does this mean from a _product_ perspective?

Reply:
Taking the KeyError as an example, would something like this be better:

raise PluginException(cause=f'An error has occurred retrieving data from the Okta API: {e}', assistance="It looks like some data for these Factors didn't have what we expected.")

I'm not sure if i should link something about KeyErrors?

Reply:
No, customer doesn't care about KeyErrors.  What you have above is better